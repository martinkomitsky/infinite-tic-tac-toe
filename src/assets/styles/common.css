@charset "UTF-8";

html {
  height: 100%;
}

body {
  margin: 0;
  height: 100%;
  /* font-family: "Arial", */
    /* sans-serif; */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  background-color: var(--color-background);
}

#infinite-tic-tac-toe {
  background-color: #f5f5f5;
  height: 100%;
  display: flex;
  /* display: grid; */
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
