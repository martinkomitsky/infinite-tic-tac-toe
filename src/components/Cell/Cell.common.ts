export const props: {
	circleValue: boolean;
	crossValue: boolean;
	hint: boolean;
	index: [number, number];
} = {
	circleValue: false,
	crossValue: true,
	hint: true,
	index: [0, 0],
};
