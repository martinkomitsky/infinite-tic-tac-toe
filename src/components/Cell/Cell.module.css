.cell {
	cursor: pointer;
	position: relative;
	width: 100px;
	height: 100px;
	display: inline-block;
	background: #eaeaea;
	margin: 3px;
	border-radius: 3px;
	transition: background-color 0.25s;
	box-shadow: 0 1px 0 0 #d7d8db, 0 0 0 1px #e3e4e8;
}

.cell:hover {
	background: #e2e2e2;
}

.cellIcon::before {
	width: 100%;
	height: 100%;
	display: flex;
	justify-content: center;
	align-items: center;
	font-size: 100px;
	text-shadow: -1px -1px 10px rgba(150, 150, 150, 0.7);
}

.cellCross::before {
	content: '×';
	position: absolute;
	color: var(--color-blue);
}

.cellCircle::before {
	content: '○';
	color: var(--color-red);
	position: absolute;
}

.cellOutlined {
	box-shadow: 0 0 0 1px #039be5;
}
