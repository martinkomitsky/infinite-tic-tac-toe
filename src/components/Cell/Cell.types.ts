export type TCellIndex = [number, number];

export interface ICellProps {
	value?: boolean;
	hint?: boolean;
	index: TCellIndex;
	onClick?: (index: TCellIndex) => void;
}
