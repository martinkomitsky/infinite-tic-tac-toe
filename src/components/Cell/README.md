# Cell

Компонент, отображающий ячейку игрового поля. Может отрисовывать крестик или нолик, может быть с "обводкой-подсказкой" хода. Имеет обработчик клика.
