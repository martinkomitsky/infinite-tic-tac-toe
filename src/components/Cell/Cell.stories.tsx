import { action } from '@storybook/addon-actions';
import { boolean, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Cell } from './Cell';

const stories = storiesOf('Infinite Tic-Tac-Toe|Cell', module).addParameters({
	readme: {
		sidebar: require('./README.md'),
	},
});

stories.addDecorator(withKnobs);

const props: {
	circleValue: boolean;
	crossValue: boolean;
	hint: boolean;
	index: [number, number];
} = {
	circleValue: false,
	crossValue: true,
	hint: true,
	index: [0, 0],
};

stories.add('initial', () => (
	<React.Fragment>
		<Cell index={props.index} onClick={action('clicked')} />
	</React.Fragment>
));

stories.add('initial with hint', () => (
	<React.Fragment>
		<Cell
			index={props.index}
			hint={boolean('Show hint', props.hint)}
			onClick={action('clicked')}
		/>
	</React.Fragment>
));

stories.add('with "X"', () => (
	<React.Fragment>
		<Cell
			index={props.index}
			value={boolean('Render "X"', props.crossValue)}
			onClick={action('clicked')}
		/>
	</React.Fragment>
));

stories.add('with "O"', () => (
	<React.Fragment>
		<Cell
			index={props.index}
			value={boolean('Render "X"', props.circleValue)}
			onClick={action('clicked')}
		/>
	</React.Fragment>
));
