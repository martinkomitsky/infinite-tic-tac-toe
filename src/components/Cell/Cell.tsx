import classnames from 'classnames';
import * as React from 'react';
import * as s from './Cell.module.css';
import * as T from './Cell.types';

export const Cell: React.FunctionComponent<T.ICellProps> = props => {
	const { value, onClick, hint, index } = props;

	const handleClick = () => {
		if (onClick) {
			onClick(index);
		}
	};

	const classNames = classnames(s.cellIcon, {
		// tslint:disable-next-line: no-boolean-literal-compare
		[s.cellCircle]: value === false,
		[s.cellCross]: value,
	});

	return (
		<div
			className={classnames(s.cell, { [s.cellOutlined]: hint })}
			onClick={handleClick}
		>
			<div className={classNames} />
		</div>
	);
};
