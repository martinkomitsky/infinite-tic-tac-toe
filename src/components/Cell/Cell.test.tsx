import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { shallow } from '../../../tests/setup/test-setup';
import { Cell } from './Cell';
import { props } from './Cell.common';

describe('<Cell />', () => {
	let wrapper: any;

	beforeEach(() => {
		wrapper = shallow(<Cell index={props.index} />);
	});

	it('should match the snapshot', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
