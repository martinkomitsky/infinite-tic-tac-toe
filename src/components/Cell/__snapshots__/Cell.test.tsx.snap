// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> should match the snapshot 1`] = `
<div
  className=""
  onClick={[Function]}
>
  <div
    className=""
  />
</div>
`;
